### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by blairlyons**
I reorganized a few things:
- moved the notebooks outside the main package, they shouldn't get installed when someone installs the package
- moved `cytosim` and `readdy` modules to the package root so they are easier to install optionally (i.e. to import from `subcell_analysis.readdy` instead of `subcell_analysis.postprocessing.readdy`), and we might add more specific functions for each of them that aren't postprocessing.
- got rid of example in the bin
- moved dependencies to main block in pyproject.toml rather than dev

Also fixed test and lint failures

### CODE REVIEW

**Review by kvegesna — APPROVED** [review]
new structure and reformatting of compression_analysis look great to me

File: subcell_analysis/compression_analysis.py [hunk 1]
```diff
@@ -1,17 +1,19 @@
 #!/usr/bin/env python
-from typing import Tuple
-import numpy as np
+from typing import Any, Tuple
```
Comment (mogres) [inline subcell_analysis/compression_analysis.py:2]: We could perhaps move this to `bin` or another subfolder like `compression_analysis`? Having a `.py` file in the main folder feels a bit out of place but it's just a personal preference/nit.

**Review by mogres — COMMENTED** [review]
Are we keeping `notebooks/actin_compression.ipynb` under version control? 
The overall reorg looks great to me! I have one small comment about potentially moving `compression_analysis.py` to its own folder but other than that we should be good to go.

**Review by mogres — APPROVED** [review]

File: subcell_analysis/compression_analysis.py (hunk 1 above)
Comment (blairlyons) [reply subcell_analysis/compression_analysis.py:2]: I think it's ok, we can move it to it's own dir later if it gets crowded.

**Comment by blairlyons** [comment]
> Are we keeping `notebooks/actin_compression.ipynb` under version control? The overall reorg looks great to me! I have one small comment about potentially moving `compression_analysis.py` to its own folder but other than that we should be good to go.

Yes the notebook is still under version control but outside the python pkg

**Pull request merged**